sway-core/src/semantic_analysis/namespace/items.rs

```diff
@@ -167,6 +170,27 @@ impl Items {
         self.implemented_traits.get_items_for_type(engines, type_id)
     }
 
+    pub fn get_impl_spans_for_decl(&self, engines: Engines<'_>, ty_decl: &TyDecl) -> Vec<Span> {
+        ty_decl
+            .return_type(engines)
+            .value
+            .map(|type_id| {
+                self.implemented_traits
+                    .get_impl_spans_for_type(engines, &type_id)
+            })
+            .unwrap_or_default()
```

Comment:
Q: Don't have the full context here, but would `unwrap_or_default()` here lead to propagating unwanted or incorrect information?

Reply:
Hmm, to add on top of Bing's comment, maybe we can consider returning `impl Iterator<Item= Span>` from these type of functions so that their consumer have more freedom to collect them or not. I guess that would also eliminate the need of using `unwrap_or_default()` as well (since an empty iterator would be returned in case of `value` being `None`).

But that being said while reading inner parts of the compiler, I tend to come across returning `Vec` as the default behaviour for these type of functions so it might not make sense to break cohesion just for these new ones. Maybe, later on, we can consider exploring a compiler api clean-up?

Reply:
Good call outs!

`unwrap_or_default` is fine here because we want it to return an empty array when there are no results. 

I think it's more important to use iterators when dealing with larger collections. Here, even in the std lib, there are typically less than 5 results.